common/core/web/input-processor/src/text/inputProcessor.ts

```diff
@@ -113,14 +117,49 @@ namespace com.keyman.text {
         // If we're performing a 'default command', it's not a standard 'typing' event - don't do fat-finger stuff.
         // Also, don't do fat-finger stuff if predictive text isn't enabled.
         if(this.languageProcessor.isActive && !ruleBehavior.triggersDefaultCommand) {
+          let keyDistribution = keyEvent.keyDistribution;
+
+          // We don't need to track absolute indexing during alternate-generation; 
+          // only position-relative, so it's better to use a sliding window for context
+          // when making alternates.  (Slightly worse for short text, matters greatly
+          // for long text.)
+          let contextWindow = new ContextWindow(preInputMock, ContextWindow.ENGINE_RULE_WINDOW);
+          let windowedMock = contextWindow.toMock();
+
           // Note - we don't yet do fat-fingering with longpress keys.
-          if(keyEvent.keyDistribution && keyEvent.kbdLayer) {
+          if(keyDistribution && keyEvent.kbdLayer) {
+            // Tracks a 'deadline' for fat-finger ops, just in case both context is long enough
+            // and device is slow enough that the calculation takes too long.
+            //
+            // Consider use of https://developer.mozilla.org/en-US/docs/Web/API/Performance/now instead?
+            // Would allow finer-tuned control.
+            let TIMEOUT_THRESHOLD = Date.now() + 16; // + 16ms.
+
+            // Tracks a minimum probability for keystroke probability.  Anything less will not be
+            // included in alternate calculations. 
+            //
+            // Seek to match SearchSpace.EDIT_DISTANCE_COST_SCALE from the predictive-text engine.
+            // Reasoning for the selected value may be seen there.  Short version - keystrokes 
+            // that _appear_ very precise may otherwise not even consider directly-neighboring keys.
+            let KEYSTROKE_EPSILON = Math.exp(-5);
+
+            // Sort the distribution into probability-descending order.
+            keyDistribution.sort(function(a, b) {
+              return b.p - a.p;
+            });
```

Comment:
```suggestion
            keyDistribution.sort((a, b) => b.p - a.p);
```

Arrow function should work here right?

Reply:
Assuming a modern-enough browser, yep.  The issue being that we've absolutely broken KMW for legacy devices / browsers in the past from this exact assumption.  Hence, I'm avoiding use of arrow functions.

Reply:
Oh I thought the typescript compiler was transpiling these down?

Reply:
Yeah, sometimes I've been surprised at what does and doesn't transpile.

Reply:
I just did a simple test and by default, tsc does transpile an arrow function down to a regular function for ES5 targets and below. Maybe we have an incorrect typescript setting?

Reply:
Huh... guess I got my wires crossed on that one.  I inspected the results of including this change, and it does indeed transpile down.